### ISSUE TITLE
Parts list aren't really formatted?

### ISSUE THREAD

**Issue by Franke123** [AlienFlight/Flight-Controllers#1]
I was thinking maybe the part lists could be formatted better, like in a spreadsheet. Here's an example I made. It has been formatted into columns and also has link and price added (this is a little biased so maybe don't include that but at least formatted columns).
[PARTS.zip](https://github.com/AlienFlight/Flight-Controllers/files/124259/PARTS.zip)

**Comment by Pedro147** [comment]
Good work Frank but can I make some suggestions please. 

- Freeze top column heading row. View > Freeze Panes > Freeze top row
- make the links clickable
- some formatting
- a formula for price tallying 
[PARTS_EDIT.xlsx](https://github.com/AlienFlight/Flight-Controllers/files/124676/PARTS_EDIT.xlsx)


Other than that a great list  I took the liberty of implementing my suggestions to see what you think but I will not get offended if you do not like the

**Comment by HumbleOne** [comment]
This is excellent work from both of you. Not everyone has Excel, so we should also leave the existing exported Parts.txt files in place. If you guys want to convert the existing Parts.txt lists (we could add them outside of the Zipped files for now) and create a pull request, I'll get these merged. :)

**Comment by Franke123** [comment]
This is awesome! Good ideas!

**Comment by Franke123** [comment]
Wow! Being a contributer to this would be amazing... You should also Pedro
to send his if he can since I won't be able to for a bit... I'll get to
work on the other FC part lists too though when I can

This is excellent work from both of you. Not everyone has Excel, so we
should also leave the existing exported Parts.txt files in place. If you
guys want to convert the existing Parts.txt lists (we could add them
outside of the Zipped files for now) and create a pull request, I'll get
these merged. :)

—
Reply to this email directly or view it on GitHub
<https://github.com/AlienFlight/Flight-Controllers/issues/1#issuecomment-182430727>
.

**Comment by 1Hammer** [comment]
Yes very nice work @Franke123 and @Pedro147. I would only make one suggestion and that would be to add a column for the labels (R1, R2, C1, C2 etc.) that indicate where the components are populated on the boards.

**Comment by Franke123** [comment]
@1Hammer that's the parts column isn't it? Its already there I think...

**Comment by 1Hammer** [comment]
Sorry @Franke123 my mistake, you did include the Parts column. I just looked at the list @Pedro147 posted and didn't see it on his. Once again very nice work from both of you!

**Comment by Franke123** [comment]
Alright cool! Thanks anyways

**Comment by madman1412** [comment]
Very good @Frank123. I see you are on your way! I will also see to it  that we will find a way to help you too! Hang tight buddy help is on the way. 
Not trying to be cryptic here so I will explain that @Franke123 is maybe in need of help in the way of procuring things to do all this as from what I understand he has limited resources. I will see what I have to spare and get a hold of you so we can get you some things that will help you in this quest.

**Comment by Franke123** [comment]
@madman1412 man I don't know what to say! I love this community! Thanks so much!

**Comment by 1Hammer** [comment]
> Very good @Frank123. I see you are on your way! I will also see to it that we will find a way to help you too! Hang tight buddy help is on the way. 
Not trying to be cryptic here so I will explain that @Franke123 is maybe in need of help in the way of procuring things to do all this as from what I understand he has limited resources. I will see what I have to spare and get a hold of you so we can get you some things that will help you in this quest.

**Comment by madman1412** [comment]
I just want to share the love @Franke123. I do not make promises I cannot keep and I can see something in you that I have seen in all of us, John, Peter, Jesse and many more. I am always happy to share with people who have a desire in this to make it something better tomorrow than it was today. One way of doing that is sharing back with others. I did not come into all this just to show people what I can do but rather show them what they can do if they really have a desire to. I know you are young in age but that makes not a bit of difference, we are all equal in my opinion and everyone has a part to play. Look forward to seeing you around and what you get into ;)

**Comment by Franke123** [comment]
@madman1412 Thats awesome of you! I think that thing you're referring to is what I call "the maker spirit", like people who'd rather know how to build something and make it themselves than buy it, regardless of the situation. I look forward to seeing you around too!

Now back to converting part lists to formatted excel documents... :smile:

**Comment by madman1412** [comment]
When you have a moment check you MMC messages @Franke123

**Comment by Pedro147** [comment]
Hi  @Franklin. 
What FC is that first spreadsheet for please, the actual "official" AlienFlight name so I can rename that spreadsheet and share it around in a few other areas, that is if you have no objections

**Comment by Franke123** [comment]
@madman1412 Sure thing!

@Pedro147 That is for the F3-V1 Quad Brushed Side USB.

**Comment by Franke123** [comment]
https://www.dropbox.com/s/6ky066dd3byth41/f3-v1partlists.zip?dl=0

Ta da! All of the F3-V1 part lists formatted all neatly. @HumbleOne Tell me when you are free for me to merge them, or you can just download this and upload it yourself. Thank you @Pedro147 for helping with formatting!

**Comment by Pedro147** [comment]
@Franke123   Great job there well done. Well what's next  :-)

**Comment by HumbleOne** [comment]
Hi @Franke123 Nice work. :smile: 
I have a question and a few more requests, before accepting a merge of these files into the project.

1.) By contributing these files, they will assume the same CC BY SA V4 that the rest of the files in this work falls under. I just want to confirm that this is ok with you before we proceed? 

2.) The f3quadpartlist appears to be using a border outline for the cells, but the other lists do not. We should probably standardize this by using or not using a cell border outline on all of the files.

3.) For the links, I think it would be easier on the eyes to utilize a short name (for text to display) instead of the full URL. Perhaps something like "Link to Purchase". The long URL can force scrolling (depending on window size and resolution), and this might be how 1Hammer missed the first Parts column.

4.) REC part link is missing the http protocol, so the link doesn't work.

**Comment by Franke123** [comment]
@HumbleOne I'll get right on it :smile:

**Comment by Pedro147** [comment]
Yes the border is just a personal preference of mine in the eventuality of the list being printed, as I think it looks better :) 
It can only be noticed when the document is in print preview or physically printed and is therefore easy to miss. 
Good call on the URL shorten @HumbleOne  
Again personal preference, but might designator be a better heading for the Part column too ?

**Comment by Franke123** [comment]
@Pedro147 I added Designator instead of part.

@HumbleOne 

1. I agree for these files to apply by the CC BY SA V4 license.
2. Added a full border outline to all of them, I don't know, it just seems more professional.
3. Added this... I'm shitty at making macro's but I finally got it to work. Also shortened the columns ofc.
4. Good call! I wouldn't have noticed this. Fixed it.


https://www.dropbox.com/s/qt650xtdo5vyutf/f3v1partlists.zip?dl=0

**Comment by Pedro147** [comment]
@Franke123  Congrats this is a really good job :alien:

**Comment by Franke123** [comment]
@Pedro147 Thanks :) It's for a good cause.

**Comment by 1Hammer** [comment]
Yes I agree with @Pedro147 , you've done an exceptional job on these @Franke123 !

**Comment by Franke123** [comment]
@1Hammer Ha thank you man I think they're pretty good.

**Comment by MJ666** [comment]
@Franke123 thanks for your work. Would you be able to make an pull request to integrate this changes? The process is relative simple if you using Eclipse. Here is an usefull tutorial: 
http://www.vogella.com/tutorials/EclipseGit/article.html
The process is mainly make an fork, get this to eclipse, create an new branch, make your updates, commit, push to git and make an pull request. I also can help with the process.

**Comment by Franke123** [comment]
@MJ666 Sure! I have never used Eclipse so you'll probably have to help me but I can do it. Maybe you could pm me on MMW on helping.

http://community.micro-motor-warehouse.com/users/fharding/

**Comment by MJ666** [comment]
@Franke123 perfect looks good. Sure there are many ways to do this.

**Comment by Franke123** [comment]
@MJ666 yeah I definitely needed an excuse to learn how to do pull requests, and this helped me lol.

**Comment by Pedro147** [comment]
@Franke123 and @MJ666 Github is certainly not the place for the faint hearted that's for sure :smile:

I suppose that I will learn how, why and what all these new terms and actions mean when, and if the time comes for me too require that knowledge. There never seems to be enough hours in the day sometimes

**Comment by Franke123** [comment]
@Pedro147 https://help.github.com/articles/github-glossary/ is super useful :smile:

**Comment by Pedro147** [comment]
Thanks @Franke123 that should get me off and racing and you most likely have worked this out already, seeing as you are pretty smart :smile: but I am Peter Newman on some of the quad forums we both frequent

**Comment by Franke123** [comment]
@Pedro147 Ha I actually didn't know that so it's good you told me... I probably should have seeing as you post Pedrowii's on your FB and your name is Pedro147.

**Comment by jamgro2434** [comment]
I know this is not the place for this, but what do you do with the board after you get it with parts soldered on? Is there a program you have to upload to the chip to tell it how to talk with and accept changes from cleanflight? (using f3 quad x)

**Comment by Franke123** [comment]
@jamgro2434 From what I understand you first have to flash the bootloader using a tool like the [USBAsp](http://www.hobbyking.com/hobbyking/store/__27990__USBasp_AVR_Programming_Device_for_ATMEL_proccessors.html) and then upload the sketch you want (whether that be multiwii, cleanflight, etc). But then again I'm no pro that's just what I think.

**Comment by jamgro2434** [comment]
hmm what would i use since i would be using the STM32F303CCT6 instead of the atmel chip

**Comment by 1Hammer** [comment]
Actually @Franke123, the new F3 boards have a BOOT tactile switch for flashing the bootloader via USB, no USBasp is required, as I understand it. It would probably be better for @jamgro2434 to start another "issue" regarding this.

**Comment by Franke123** [comment]
@1Hammer Ah, thank you very much! Didn't know that :P

**Comment by jamgro2434** [comment]
[What to do After Building](https://github.com/AlienFlight/Flight-Controllers/issues/4)
